lib/spack/spack/spec.py

```diff
@@ -2232,14 +2232,25 @@ def concretize(self, tests=False):
             mvar._patches_in_order_of_appearance = list(
                 t[-1] for t in ordered_hashes)
 
+        # This is awful!!!!
+        # TODO: anything better
+        # This allows us to access the package class for externals that
+        # override their prefix. We then mark the spec as concrete again after
+        # setting the external paths.
+        self._mark_concrete()
+
         for s in self.traverse():
             if s.external_module and not s.external_path:
                 compiler = spack.compilers.compiler_for_spec(
                     s.compiler, s.architecture)
                 for mod in compiler.modules:
                     md.load_module(mod)
 
-                s.external_path = md.get_path_from_module(s.external_module)
+                try:
+                    s.external_path = s.package.external_prefix
+                except AttributeError:
```

Comment:
Prefer to use `.get()` instead of catching `AttributeError`.  `AttributeError` can easily catch the wrong thing (any `.` above) and you can miss a bug entirely.